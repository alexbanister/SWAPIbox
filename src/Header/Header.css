.header {
  background: #211E27;
  background: -moz-linear-gradient(top, #211E27 0%, #172633 100%);
  background: -webkit-linear-gradient(top, #211E27 0%, #172633 100%);
  background: linear-gradient(to bottom, #211E27 0%, #172633 100%);
  filter: progid:DXImageTransform.Microsoft.gradient( startColorstr='$color1', endColorstr='$color2',GradientType=0 );
  display: flex;
  flex-direction: column;
  align-items: center;
  border-bottom: 2px solid #C43B41; }
  .header img {
    display: block;
    width: 50vw;
    padding: 5em; }
  .header nav {
    padding-bottom: 2em;
    font-weight: 300; }
    .header nav a {
      text-decoration: none;
      color: #6D6D74;
      padding: .5em;
      margin: .5em; }
      .header nav a:hover {
        color: #C43B41; }
    .header nav .active {
      font-weight: 900;
      color: #C43B41; }
