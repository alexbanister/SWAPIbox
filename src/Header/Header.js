import React from 'react';
import PropTypes from 'prop-types';
import Button from '../Button/Button';
import logo from '../assets/logo.svg'

const Header = () => {
  return (
    <div className='header'>
      <a href="/"><img src={logo} alt='logo' /></a>
      <nav>
        <Button path='/people' text='People' />
        <Button path='/planets' text='Planets' />
        <Button path='/vehicles' text='Vehicles' />
        <Button path='/favorites' text='Favorites' />
      </nav>
    </div>
  );
};

Header.propTypes = {
  compareInfo: PropTypes.objectOf(PropTypes.object),
  removeCompare: PropTypes.func
};

export default Header;
