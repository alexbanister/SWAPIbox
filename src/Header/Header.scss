@import "../variables.scss";

.header {
  @include linear-gradient($dark-grey, $light-grey);
  display: flex;
  flex-direction: column;
  align-items: center;
  border-bottom: 2px solid $highlight-dark;
  img {
    display: block;
    width: 50vw;
    padding: 3em;
  }
  nav {
    padding: 2em;
    font-weight: 300;
    a {
      text-decoration: none;
      color: $body-light;
      padding: .5em;
      margin: .5em;
      &:hover {
        color: $highlight-dark;
      }
    }
    .active {
      font-weight: 900;
    }
  }


}
