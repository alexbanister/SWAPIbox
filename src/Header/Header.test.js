import React from 'react';
import { shallow, mount } from 'enzyme';
import Header from './Header';



describe('Header', () => {
  it('should render correctly and match the snapshot', () => {
    const mockFunc = jest.fn();
    const wrapper = shallow(<Header favCount={mockFunc} />);

    expect(wrapper).toMatchSnapshot();
  })
})




