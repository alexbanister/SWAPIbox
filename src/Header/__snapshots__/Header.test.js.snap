// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header should render correctly and match the snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Header
    favCount={[Function]}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": null,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <a
          href="/"
>
          <img
                    alt="logo"
                    src="logo.svg"
          />
</a>,
        <nav>
          <NavLink
                    activeClassName="active"
                    ariaCurrent="true"
                    to="/people"
          >
                    People
          </NavLink>
          <NavLink
                    activeClassName="active"
                    ariaCurrent="true"
                    to="/planets"
          >
                    Planets
          </NavLink>
          <NavLink
                    activeClassName="active"
                    ariaCurrent="true"
                    to="/vehicles"
          >
                    Vehicles
          </NavLink>
          <NavLink
                    activeClassName="active"
                    ariaCurrent="true"
                    to="/favorites"
          >
                    Favorites 
                    <i>
                              (
                              [Function]
                              )
                    </i>
          </NavLink>
</nav>,
      ],
      "className": "header",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": null,
        "nodeType": "host",
        "props": Object {
          "children": <img
            alt="logo"
            src="logo.svg"
/>,
          "href": "/",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": null,
          "nodeType": "host",
          "props": Object {
            "alt": "logo",
            "src": "logo.svg",
          },
          "ref": null,
          "rendered": null,
          "type": "img",
        },
        "type": "a",
      },
      Object {
        "instance": null,
        "key": null,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <NavLink
              activeClassName="active"
              ariaCurrent="true"
              to="/people"
>
              People
</NavLink>,
            <NavLink
              activeClassName="active"
              ariaCurrent="true"
              to="/planets"
>
              Planets
</NavLink>,
            <NavLink
              activeClassName="active"
              ariaCurrent="true"
              to="/vehicles"
>
              Vehicles
</NavLink>,
            <NavLink
              activeClassName="active"
              ariaCurrent="true"
              to="/favorites"
>
              Favorites 
              <i>
                            (
                            [Function]
                            )
              </i>
</NavLink>,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": null,
            "nodeType": "function",
            "props": Object {
              "activeClassName": "active",
              "ariaCurrent": "true",
              "children": "People",
              "to": "/people",
            },
            "ref": null,
            "rendered": "People",
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": null,
            "nodeType": "function",
            "props": Object {
              "activeClassName": "active",
              "ariaCurrent": "true",
              "children": "Planets",
              "to": "/planets",
            },
            "ref": null,
            "rendered": "Planets",
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": null,
            "nodeType": "function",
            "props": Object {
              "activeClassName": "active",
              "ariaCurrent": "true",
              "children": "Vehicles",
              "to": "/vehicles",
            },
            "ref": null,
            "rendered": "Vehicles",
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": null,
            "nodeType": "function",
            "props": Object {
              "activeClassName": "active",
              "ariaCurrent": "true",
              "children": Array [
                "Favorites ",
                <i>
                  (
                  [Function]
                  )
</i>,
              ],
              "to": "/favorites",
            },
            "ref": null,
            "rendered": Array [
              "Favorites ",
              Object {
                "instance": null,
                "key": null,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    "(",
                    [Function],
                    ")",
                  ],
                },
                "ref": null,
                "rendered": Array [
                  "(",
                  [Function],
                  ")",
                ],
                "type": "i",
              },
            ],
            "type": [Function],
          },
        ],
        "type": "nav",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": null,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <a
            href="/"
>
            <img
                        alt="logo"
                        src="logo.svg"
            />
</a>,
          <nav>
            <NavLink
                        activeClassName="active"
                        ariaCurrent="true"
                        to="/people"
            >
                        People
            </NavLink>
            <NavLink
                        activeClassName="active"
                        ariaCurrent="true"
                        to="/planets"
            >
                        Planets
            </NavLink>
            <NavLink
                        activeClassName="active"
                        ariaCurrent="true"
                        to="/vehicles"
            >
                        Vehicles
            </NavLink>
            <NavLink
                        activeClassName="active"
                        ariaCurrent="true"
                        to="/favorites"
            >
                        Favorites 
                        <i>
                                    (
                                    [Function]
                                    )
                        </i>
            </NavLink>
</nav>,
        ],
        "className": "header",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": null,
          "nodeType": "host",
          "props": Object {
            "children": <img
              alt="logo"
              src="logo.svg"
/>,
            "href": "/",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": null,
            "nodeType": "host",
            "props": Object {
              "alt": "logo",
              "src": "logo.svg",
            },
            "ref": null,
            "rendered": null,
            "type": "img",
          },
          "type": "a",
        },
        Object {
          "instance": null,
          "key": null,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <NavLink
                activeClassName="active"
                ariaCurrent="true"
                to="/people"
>
                People
</NavLink>,
              <NavLink
                activeClassName="active"
                ariaCurrent="true"
                to="/planets"
>
                Planets
</NavLink>,
              <NavLink
                activeClassName="active"
                ariaCurrent="true"
                to="/vehicles"
>
                Vehicles
</NavLink>,
              <NavLink
                activeClassName="active"
                ariaCurrent="true"
                to="/favorites"
>
                Favorites 
                <i>
                                (
                                [Function]
                                )
                </i>
</NavLink>,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": null,
              "nodeType": "function",
              "props": Object {
                "activeClassName": "active",
                "ariaCurrent": "true",
                "children": "People",
                "to": "/people",
              },
              "ref": null,
              "rendered": "People",
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": null,
              "nodeType": "function",
              "props": Object {
                "activeClassName": "active",
                "ariaCurrent": "true",
                "children": "Planets",
                "to": "/planets",
              },
              "ref": null,
              "rendered": "Planets",
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": null,
              "nodeType": "function",
              "props": Object {
                "activeClassName": "active",
                "ariaCurrent": "true",
                "children": "Vehicles",
                "to": "/vehicles",
              },
              "ref": null,
              "rendered": "Vehicles",
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": null,
              "nodeType": "function",
              "props": Object {
                "activeClassName": "active",
                "ariaCurrent": "true",
                "children": Array [
                  "Favorites ",
                  <i>
                    (
                    [Function]
                    )
</i>,
                ],
                "to": "/favorites",
              },
              "ref": null,
              "rendered": Array [
                "Favorites ",
                Object {
                  "instance": null,
                  "key": null,
                  "nodeType": "host",
                  "props": Object {
                    "children": Array [
                      "(",
                      [Function],
                      ")",
                    ],
                  },
                  "ref": null,
                  "rendered": Array [
                    "(",
                    [Function],
                    ")",
                  ],
                  "type": "i",
                },
              ],
              "type": [Function],
            },
          ],
          "type": "nav",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
